packages/codecs-strings/src/baseX.ts

```diff
@@ -92,9 +92,12 @@ export const getBaseXDecoder = (alphabet: string): VariableSizeDecoder<string> =
 export const getBaseXCodec = (alphabet: string): VariableSizeCodec<string> =>
     combineCodec(getBaseXEncoder(alphabet), getBaseXDecoder(alphabet));
 
-function partitionLeadingZeroes(value: string, zeroCharacter: string): [string, string] {
-    const leadingZeroIndex = [...value].findIndex(c => c !== zeroCharacter);
-    return leadingZeroIndex === -1 ? [value, ''] : [value.slice(0, leadingZeroIndex), value.slice(leadingZeroIndex)];
+function partitionLeadingZeroes(
+    value: string,
+    zeroCharacter: string,
+): [leadingZeros: string, tailChars: string | undefined] {
+    const [leadingZeros, tailChars] = value.split(new RegExp(`((?!${zeroCharacter}).*)`));
```

Comment:
This regular expression says ‘split on the first non-zeroCharacter and onward.’ This naturally gives us a partition, which we can output directly.

Reply:
Nice! Should we just return that line directly instead of allocating a new array?

Reply:
Annoyingly, the return is `[leading, tailing, emptyString]`.